### PR TITLE
Better support for wildcard certificates:

### PR DESCRIPTION
**Pull request by JeetKunDoug**
- Update make_certs to support wildcard certificates by passing an additional Path parameter.
- add replication2_ssl test to ensure using wildcard certificates works after PR to riak_core_ssl_util.

### CODE REVIEW

**Comment by JeetKunDoug** [comment]
Depends on https://github.com/basho/riak_core/pull/701, which is being held until 2.0.5 is released, so please DON'T MERGE yet.

File: src/make_certs.erl [hunk 1]
```diff
@@ -112,8 +112,14 @@ intermediateCA(Root, CA, ParentCA) ->
 endusers(Root, CA, Users) ->
     [enduser(Root, CA, User) || User <- Users].
 
-enduser(Root, CA, User) -> 
-    UsrRoot = filename:join([Root, User]),
+enduser(Root, CA, {User, Path}) ->
```
Comment (tburghart) [inline src/make_certs.erl]: Minor, but I'd prefer if the `Path` parameter was named something like `DirName`, as it's only one segment of the filesystem path.

File: src/make_certs.erl [hunk 2]
```diff
@@ -112,8 +112,14 @@ intermediateCA(Root, CA, ParentCA) ->
...
+    enduser(Root, CA, User, Path);
+
+enduser(Root, CA, User) ->
+    enduser(Root, CA, User, User).
+
+enduser(Root, CA, User, Path) ->
```
Comment (tburghart) [inline src/make_certs.erl]: Same comment re: name of `Path` parameter.

File: tests/replication2_ssl.erl [hunk 3]
```diff
@@ -188,6 +189,25 @@ confirm() ->
...
+    SSLConfig8 = [
+        {riak_repl,
+            [
+                {fullsync_on_connect, false},
+                {fullsync_interval, disabled}
+            ]},
+        {riak_core,
+            [
+                {ssl_enabled, true},
+                {peer_common_name_acl, ["*.basho.com"]},
```
Comment (tburghart) [inline tests/replication2_ssl.erl]: Simple wildcard test should not use an ACL.

File: tests/replication2_ssl.erl [hunk 4]
```diff
@@ -253,6 +273,12 @@ confirm() ->
     ?assertEqual(ok, test_connection({Node1, merge_config(SSLConfig5, BaseConf)},
             {Node2, merge_config(SSLConfig6, BaseConf)})),
 
+
+    lager:info("===testing using same wildcard certificate"),
+    rt:log_to_nodes([Node1, Node2], "wildcard certificates test"),
+    ?assertEqual(ok, test_connection({Node1, merge_config(SSLConfig8, BaseConf)},
+        {Node2, merge_config(SSLConfig8, BaseConf)})),
+
```
Comment (tburghart) [inline tests/replication2_ssl.erl]: Should have a test connecting wildcarded cert to non-wildcarded cert, without ACL, such as `SSLConfig3` <==> `SSLConfig8`

File: tests/replication2_ssl.erl (hunk 3 above)
Comment (JeetKunDoug) [inline tests/replication2_ssl.erl]: Not sure why GitHub left this as a "valid" comment - SSLConfig8 now has no ACL, and I added 9 to make sure we tested wildcard certs with ACLs

**Comment by tburghart** [comment]
+1 137a11c

**Pull request merged**